llvm/include/llvm/SYCLLowerIR/CleanupSYCLCompilerInternalMetadata.h

```diff
@@ -0,0 +1,31 @@
+//===- CleanupSYCLCompilerInternalMetadata.h - SYCLRemoveAspectsMD Pass --===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Cleanup SYCL compiler internal metadata (usually inserted by sycl-post-link)
+// as it will never be used in the compilation ever again
+//
+//===----------------------------------------------------------------------===//
+//
+#ifndef LLVM_CLEANUP_SYCL_COMPILER_INTERNAL_METADATA
+#define LLVM_CLEANUP_SYCL_COMPILER_INTERNAL_METADATA
```

Comment:
Can we use `#pragma once`, some of the other headers in this folder are using it.

Reply:
we indeed can, but from what I can tell guarding macro is more standard practice in LLVM. For sure I won't fight for this and can change if you believe it's better :)

Reply:
I don't think that we should use `#pragma once` in here: [LLVM Coding Standards](https://llvm.org/docs/CodingStandards.html#c-standard-versions):

> Unless otherwise documented, LLVM subprojects are written using standard C++17 code and avoid unnecessary vendor-specific extensions.

`#pragma once` is **non-standard** C++ functionality, even though widely supported.

I don't see any mentions of it neither in the Coding Standards, nor in [Programmer's Manual](https://llvm.org/docs/ProgrammersManual.html), but there is [Header Guard](https://llvm.org/docs/CodingStandards.html#header-guard) section:

> The header file’s guard should be the all-caps path that a user of this header would `#include`, using ‘_’ instead of path separator and extension marker. For example, the header file `llvm/include/llvm/Analysis/Utils/Local.h` would be `#include`-ed as `#include "llvm/Analysis/Utils/Local.h"`, so its guard is `LLVM_ANALYSIS_UTILS_LOCAL_H`.

Which suggests that macro include guard is the way to go

Reply:
Unfortunate, but yeah seems we should use macros, thanks